Mathlib/Algebra/Algebra/Equiv.lean

```diff
@@ -318,13 +369,16 @@ theorem coe_refl : ⇑(refl : A₁ ≃ₐ[R] A₁) = id :=
 
 /-- Algebra equivalences are symmetric. -/
 @[symm]
-def symm (e : A₁ ≃ₐ[R] A₂) : A₂ ≃ₐ[R] A₁ :=
+def symm : A₂ ≃ₐ[R] A₁ :=
   { e.toRingEquiv.symm with
-    commutes' := fun r => by
-      rw [← e.toRingEquiv.symm_apply_apply (algebraMap R A₁ r)]
+    map_smul' := fun r a => by
```

Comment:
It's a shame you can't use some results about `LinearEquiv` here; I assume semilinearity puts a spanner in the works for doing an analogous generalization there.

Reply:
Yeah, it would introduce type class assumptions I would prefer to avoid due to semilinearity. This is likely golfable though.